src/NuGet.Core/NuGet.PackageManagement/NetStandard20CompatibilityUtil.cs

```diff
@@ -0,0 +1,145 @@
+                    && nearestFramework.Version >= FrameworkConstants.CommonFrameworks.NetStandard20.Version)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        internal static async Task InstallNetStandard20CompatibilityPackage(NuGetProject nuGetProject,
+            INuGetProjectContext nuGetProjectContext,
+            NuGetPackageManager packageManager,
+            ISourceRepositoryProvider sourceRepositoryProvider,
+            CancellationToken token)
+        {
+            // First check if the compatibility package is already installed.
+            var projectInstalledPackageReferences = await nuGetProject.GetInstalledPackagesAsync(token);
+
+            var installedPackageReference = projectInstalledPackageReferences
+                .Where(pr => StringComparer.OrdinalIgnoreCase.Equals(pr.PackageIdentity.Id, CompatibilityPackageId))
+                .FirstOrDefault();
+
+            if (installedPackageReference != null)
+            {
+                return;
+            }
+
+            nuGetProjectContext.Log(
+                    ProjectManagement.MessageLevel.Info,
+                    Strings.InstallingNetstandard20CompatibilityPackage);
+
+            var resolutionContext = new ResolutionContext(DependencyBehavior.Lowest,
+            includePrelease: false,
+            includeUnlisted: false,
+            versionConstraints: VersionConstraints.None);
+
+            var primarySources = sourceRepositoryProvider.GetRepositories()
+                .Where(e => e.PackageSource.IsEnabled);
+
+            var log = new LoggerAdapter(nuGetProjectContext);
+
+            // Check for latest stable version of the package.
+            var resolvedPackage = await NuGetPackageManager.GetLatestVersionAsync(CompatibilityPackageId,
+                nuGetProject,
+                resolutionContext,
+                primarySources,
+                log,
+                token);
+
+            if (resolvedPackage?.LatestVersion == null)
+            {
+                // If no stable version of the package could be found, then look for latest pre-release.
+                resolutionContext = new ResolutionContext(DependencyBehavior.Lowest,
+                    includePrelease: true,
+                    includeUnlisted: false,
+                    versionConstraints: VersionConstraints.None);
+
+                resolvedPackage = await NuGetPackageManager.GetLatestVersionAsync(CompatibilityPackageId,
+                    nuGetProject,
+                    resolutionContext,
+                    primarySources,
+                    log,
+                    token);
+
+                if (resolvedPackage == null || resolvedPackage.LatestVersion == null)
+                {
+                    throw new InvalidOperationException(string.Format(Strings.NoLatestVersionFound, CompatibilityPackageId));
```

Comment:
Consider warning and skipping, this could get triggered during file new project or some other template scenario. In those cases the caller is often limited to only providing one source at a time.